toolchain/check/inst.cpp

```diff
@@ -40,7 +43,8 @@ static auto FinishInst(Context& context, SemIR::InstId inst_id,
 
   // Keep track of dependent instructions.
   if (dep_kind != GenericRegionStack::DependencyKind::None) {
-    // TODO: Also check for template-dependent instructions.
+    // Template-dependent instructions are handled separately by
+    // `AddDependentActionInst`.
```

Comment:
Is there any checking that we are correctly routing template-dependent instructions?

Reply:
I've added a `CHECK` for this.